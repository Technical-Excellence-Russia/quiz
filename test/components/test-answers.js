import React from "react"
import renderer from "react-test-renderer"

import Context from "../../src/context"
import Answers from "../../src/components/answers"

jest.mock("../../src/components/answer", () => (props) => {
	expect(props).toEqual({index: "0", answer: {name: "X"}})
	return "Answer"
})

describe("Answer should", () => {

	let component = null
	const state = {question: {answers: [{name: "X"}]}}

	beforeAll(() => component = renderer.create(
		<Context.Provider value={[state]}>
			<Answers/>
		</Context.Provider>)
	)

	test("equal to snapshot", () => {
		expect(component.toJSON()).toMatchSnapshot()
		expect.assertions(2)
	})

	afterAll(() => component.unmount())
})