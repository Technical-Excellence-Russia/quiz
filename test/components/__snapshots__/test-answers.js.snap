// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answer should equal to snapshot 1`] = `"Answer"`;
